runtime/logger/logger.go

```diff
@@ -112,3 +114,10 @@ func NewLogger(opts Options) logr.Logger {
 
 	return zap.New(zap.UseFlagOptions(&zapOpts))
 }
+
+// SetLogger sets the logger for the controller-runtime and klog packages to the given logger.
+// It is not thread-safe, and should be called as early as possible in the program's execution.
+func SetLogger(logger logr.Logger) {
+	ctrl.SetLogger(logger)
+	klog.SetLoggerWithOptions(logger, klog.ContextualLogger(true))
```

Comment:
Should we set a name on this logger?

Reply:
That's not related to anything this is doing? The `setup` log is used in `main`, this sets however the general logging function being utilized by `klog` calls, which at present would result in it logging something like:

```json
{"level":"info","ts":"2023-03-07T12:22:54.290Z","msg":"attempting to acquire leader lease flux-system/helm-controller-leader-election...\n"}
{"level":"info","ts":"2023-03-07T12:22:54.297Z","msg":"successfully acquired lease flux-system/helm-controller-leader-election\n"}
```

Or, for underlying (discovery) client calls:

```json
{"level":"error","ts":"2023-03-07T12:23:21.825Z","msg":"couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request\n","stacktrace":"k8s.io/client-go/discovery.fetchGroupVersionResources.func1\n\tk8s.io/client-go@v0.26.2/discovery/discovery_client.go:520"}
```

This could be sugared with e.g. `"logger":"runtime"` (or `"logger":"klog", or whatnot) by doing:

```go
klog.SetLoggerWithOptions(logger.WithName("runtime"), klog.ContextualLogger(true))
```

Reply:
Ok, runtime feels like a good name